### PR TITLE
D8CORE-2017: Integrate stanford_news and update default_content for n…

### PR DESCRIPTION
**Pull request by sherakama**
…o ids.

# READY FOR REVIEW/NOT READY
- (Edit the above to reflect status)

# Summary
- TL;DR - what's this PR for?

# Needed By (Date)
- When does this need to be merged by?

# Urgency
- How critical is this PR?

# Steps to Test

1. Do this
1. Then this
2. Then this

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: config/sync/field.storage.node.layout_builder__layout.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-uuid: 833a0790-a592-43cb-90d6-f15509b9c74a
+uuid: ed8c3733-5cf1-4e6d-940e-350e7c8d76b8
```
Comment (pookmish) [inline config/sync/field.storage.node.layout_builder__layout.yml:1]: We cant change the uuid of existing config

Comment (sherakama) [reply config/sync/field.storage.node.layout_builder__layout.yml:1]: Good catch. I'll look for more of these.

**Comment by sherakama** [comment]
> does this have the cleanup of config from news and people that was done? there were some configs in those modules I removed.

I'll sync these up again to be sure.

> Also can we configure node revision delete in this.
Oh I think I did that on the SOE Profile but not this. Let me confirm.

**Comment by sherakama** [comment]
Node revision delete is set up: https://accelerants20200514-dev.sites.stanford.edu/admin/structure/types/manage/stanford_news

**Comment by pookmish** [comment]
I did a review of all the new config files and cross referenced them to the news and person modules. why aren't some of these configs included in those modules:
**News**
- core.base_field_override.node.stanford_news.promote.yml
- metatag.metatag_defaults.node__stanford_news.yml
- metatag.metatag_defaults.page_variant__stanford_news_list-layout_builder-0.yml
- metatag.metatag_defaults.page_variant__stanford_news_list_terms-layout_builder-0.yml
- xmlsitemap.settings.node.stanford_news.yml
- xmlsitemap.settings.taxonomy_term.stanford_news_topics.yml

**Person**
- metatag.metatag_defaults.node__stanford_person.yml
- metatag.metatag_defaults.page_variant__people-layout_builder-0.yml
- metatag.metatag_defaults.page_variant__stanford_person_list-layout_builder-1.yml
- rabbit_hole.behavior_settings.node_type_stanford_person.yml
- xmlsitemap.settings.node.stanford_person.yml
- xmlsitemap.settings.taxonomy_term.stanford_person_types.yml

These seem to be desirable configs within the module as well.  yes it adds another dependency on metatag and xmlsitemap modules, but thats not a bad thing.

File: config/sync/field.field.node.stanford_page.su_page_components.yml [hunk 2]
```diff
@@ -32,13 +32,18 @@ settings:
...
+        enabled: false
       stanford_media_caption:
         weight: 8
         enabled: false
       stanford_spacer:
         weight: 9
         enabled: false
     widths:
+      news_views:
+        min: '12'
```
[the remark this answers is not in the document]
Comment (sherakama) [reply config/sync/field.field.node.stanford_page.su_page_components.yml:46]: The display for the rows and columns is handled in the view itself. No need to put things side x side.

[the remark this answers is not in the document]
Comment (pookmish) [reply config/sync/field.field.node.stanford_page.su_page_components.yml:46]: you know there will be a desire for a view like that in the future. like a "most recent" news block. or a list of news to the side of events/people. but we can always adjust at that time

**Comment by sherakama** [comment]
22:08
Shea(opens in new tab) I was trying to keep the dependencies down for the most part
22:09
pookmish(opens in new tab) well we could just dump them into config/optional :shrug_guy:
22:09
Shea(opens in new tab) I suppose we could and it won't really hurt
22:10
pookmish(opens in new tab) yeah. i think the core.base_field_override.node.stanford_news.promote.yml should go in install dir. but the rest can be optional
22:10
Shea(opens in new tab) That is a good question on something like that one
22:10
Do we want to force that field to be off for everyone?
22:10
None of the views that we package use promote to front
22:10
But somebody may?
22:12
pookmish(opens in new tab) its just an 'install' state anyways. it doesn't stop other sites from enabling it (edited)

**Comment by pookmish** [comment]
https://github.com/SU-SWS/stanford_person/pull/87
https://github.com/SU-SWS/stanford_news/pull/74

**Comment by sherakama** [comment]
@pookmish all yours. Good night.

File: tests/codeception/acceptance/Content/PersonCest.php [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+   */
+  public function testCreatePerson(AcceptanceTester $I) {
+    $I->createEntity([
+      'type' => 'stanford_person',
+      'su_person_first_name' => "John",
+      'su_person_last_name' => "Wick",
+    ]);
+    $I->amOnPage("/person/john-wick");
+    $I->see("John Wick");
+    $I->runDrush('cr');
```
Comment (pookmish) [inline tests/codeception/acceptance/Content/PersonCest.php:56]: I've noticed that when you use the `$I->createEntity()` it won't clear caches. but if you go through the form to create the node, it will. and then you don't need to clear cache.

File: tests/codeception/acceptance/Content/NewsCest.php [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+  /**
+   * Test that the default content has installed and is unpublished.
+   */
+  public function testDefaultContentExists(AcceptanceTester $I) {
+    $I->logInWithRole('administrator');
+    $I->amOnPage("/admin/content");
+    $I->see("Sample: Smith Conference");
+    $I->see("Sample: For Runners, Is 15 Feet the New 6 Feet for Social Distancing?");
+    $I->see("Sample: Stanford researchers find that misfiring from jittery neurons");
```
Comment (pookmish) [inline tests/codeception/acceptance/Content/NewsCest.php:16]: when we get alot of default content, these might be on a different pager.
something we might have to deal with later

Comment (sherakama) [reply tests/codeception/acceptance/Content/NewsCest.php:16]: yeah.

File: tests/codeception/acceptance/Content/PersonCest.php (hunk 3 above)
Comment (sherakama) [reply tests/codeception/acceptance/Content/PersonCest.php:56]: Good to know.

**Comment by pookmish** [comment]
Ok i'm going to do a complete local rebuild with this and do a once over
